tests/functional/ovndbcluster_controller_test.go

```diff
@@ -0,0 +1,277 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package functional_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/google/uuid"
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	condition "github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	. "github.com/openstack-k8s-operators/lib-common/modules/test/helpers"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe("OVNDBCluster controller", func() {
+
+	var namespace string
+	var OVNDBClusterName types.NamespacedName
+
+	BeforeEach(func() {
+		// NOTE(gibi): We need to create a unique namespace for each test run
+		// as namespaces cannot be deleted in a locally running envtest. See
+		// https://book.kubebuilder.io/reference/envtest.html#namespace-usage-limitation
+		namespace = uuid.New().String()
+		th.CreateNamespace(namespace)
+		// We still request the delete of the Namespace to properly cleanup if
+		// we run the test in an existing cluster.
+		DeferCleanup(th.DeleteNamespace, namespace)
+		// lib-common uses OPERATOR_TEMPLATES env var to locate the "templates"
+		// directory of the operator. We need to set them otherwise lib-common
+		// will fail to generate the ConfigMap as it does not find ovn-dbcluster.json
+		err := os.Setenv("OPERATOR_TEMPLATES", "../../templates")
+		Expect(err).NotTo(HaveOccurred())
+
+		name := fmt.Sprintf("ovndbcluster-%s", uuid.New().String())
+		instance := CreateOVNDBCluster(namespace, name, GetDefaultOVNDBClusterSpec())
+		OVNDBClusterName = types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()}
+		DeferCleanup(DeleteInstance, instance)
+	})
+
+	When("A OVNDBCluster instance is created", func() {
+
+		It("should have the Spec fields initialized", func() {
+			OVNDBCluster := GetOVNDBCluster(OVNDBClusterName)
+			Expect(OVNDBCluster.Spec.Replicas).Should(Equal(int32(1)))
+			Expect(OVNDBCluster.Spec.LogLevel).Should(Equal("info"))
+			Expect(OVNDBCluster.Spec.DBType).Should(Equal("NB"))
+		})
+
+		It("should have the Status fields initialized", func() {
+			OVNDBCluster := GetOVNDBCluster(OVNDBClusterName)
+			Expect(OVNDBCluster.Status.Hash).To(BeEmpty())
+			Expect(OVNDBCluster.Status.ReadyCount).To(Equal(int32(0)))
+		})
+
+		It("should have a finalizer", func() {
+			// the reconciler loop adds the finalizer so we have to wait for
+			// it to run
+			Eventually(func() []string {
+				return GetOVNDBCluster(OVNDBClusterName).Finalizers
+			}, timeout, interval).Should(ContainElement("OVNDBCluster"))
+		})
+
+		It("should not create a config-data config map", func() {
```

Comment:
This looks like the script and config-data checks can be combined into one table test, for example, as shown here: https://github.com/openstack-k8s-operators/nova-operator/blob/master/test/functional/nova_reconfiguration_test.go#L324